api/src/main/java/org/openmrs/module/fgh/mpi/MpiContext.java

```diff
@@ -0,0 +1,248 @@
+		if (!this.contextInitialized) {
+			AdministrationService adminService = Context.getAdministrationService();
+			
+			String gpAuthenticationType = adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE);
+			
+			if (StringUtils.isBlank(gpAuthenticationType)) {
+				throw new APIException(GP_AUTHENTICATION_TYPE + " global property value is not set");
+			}
+			
+			this.authenticationType = AuthenticationType.valueOf(gpAuthenticationType);
+			
+			if (log.isDebugEnabled()) {
+				log.debug("Authentication Type: " + this.authenticationType);
+			}
+			
+			this.serverBaseUrl = adminService.getGlobalProperty(GP_MPI_BASE_URL);
+			if (StringUtils.isBlank(this.serverBaseUrl)) {
+				throw new APIException(GP_MPI_BASE_URL + " global property value is not set");
+			}
+			
+			if (log.isDebugEnabled()) {
+				log.debug("The MPI Syetem base server URL: " + this.serverBaseUrl);
+			}
+			
+			String gpMpiSystem = adminService.getGlobalProperty(GP_MPI_SYSTEM);
+			
+			if (StringUtils.isBlank(gpMpiSystem)) {
+				throw new APIException(GP_MPI_SYSTEM + " global property value is not set");
+			}
+			
+			this.mpiSystem = MpiSystemType.valueOf(gpMpiSystem);
+			
+			if (log.isDebugEnabled()) {
+				log.debug("MPI system: " + gpMpiSystem);
+			}
+			
+			this.openmrsUuidSystem = MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM);
+			
+			if (StringUtils.isBlank(this.openmrsUuidSystem)) {
+				throw new APIException(GP_UUID_SYSTEM + " global property value is not set");
+			}
+			
+			if (log.isDebugEnabled()) {
+				log.debug("Openmrs UUID System: " + this.openmrsUuidSystem);
+			}
+			
+			this.contentType = adminService.getGlobalProperty(GP_MPI_APP_CONTENT_TYPE);
+			if (StringUtils.isBlank(this.serverBaseUrl)) {
+				throw new APIException(GP_MPI_APP_CONTENT_TYPE + " global property value is not set");
+			}
+			
+			if (log.isDebugEnabled()) {
+				log.debug("Content Type: " + this.contentType);
+			}
+			
+			if (this.authenticationType.isCertificate()) {
+				initSSL();
+			} else if (this.authenticationType.isOuath()) {
+				initOauth();
+			}
+			
+			this.contextInitialized = true;
+		}
+	}
+	
+	protected void initOauth() {
+		AdministrationService adminService = Context.getAdministrationService();
+		
+		this.clientId = adminService.getGlobalProperty(GP_SANTE_CLIENT_ID);
+		
+		if (StringUtils.isBlank(this.clientId)) {
+			throw new APIException(GP_SANTE_CLIENT_ID + " global property value is not set");
+		}
+		
+		if (log.isDebugEnabled()) {
+			log.debug("Client Id: " + this.clientId);
+		}
+		
+		this.clientSecret = adminService.getGlobalProperty(GP_SANTE_CLIENT_SECRET);
+		
+		if (StringUtils.isBlank(this.clientSecret)) {
+			throw new APIException(GP_SANTE_CLIENT_SECRET + " global property value is not set");
+		}
+		
+		if (log.isDebugEnabled()) {
+			log.debug("Client Secret: " + this.clientSecret);
+		}
+	}
+	
+	protected void initSSL() throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException,
```

Comment:
Usually if a method has more than 2 or 3 exception types in the throws clause, I find their first common exception type and use that only